lib/prometheus_exporter/instrumentation/sidekiq.rb

```diff
@@ -9,21 +9,42 @@ def initialize(client: nil)
 
     def call(worker, msg, queue)
       success = false
+      shutdown = false
       start = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
       result = yield
       success = true
       result
+    rescue ::Sidekiq::Shutdown => e
+      shutdown = true
+      raise e
     ensure
       duration = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC) - start
       class_name = worker.class.to_s == 'ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper' ?
                      msg['wrapped'] : worker.class.to_s
-
       @client.send_json(
         type: "sidekiq",
         name: class_name,
         success: success,
+        shutdown: shutdown,
+        retries_exhausted: retries_exhausted?(msg),
         duration: duration
       )
     end
+
+    private
+
+    def retries_exhausted?(msg)
+      max_retries_or_bool = msg["retry"]
+      return true unless max_retries_or_bool
+      max_retries =
+        if max_retries_or_bool.is_a?(Numeric)
+          max_retries_or_bool
+        else
+          #Sidekiq's default max_retries is 25
+          ::Sidekiq.options[:max_retries] || 25
```

Comment:
You can document this more effectively using a constant than a comment, e.g. `::Sidekiq.options[:max_retries] || SIDEKIQ_DEFAULT_MAX_RETRIES` but it would be much more effective to use the value directly from Sidekiq if you can find a constant (e.g. `Sidekiq::Config::DEFAULT_MAX_RETRIES`)

Reply:
Please use https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/job_retry.rb#L63 or even better validate if you can use https://www.rubydoc.info/gems/sidekiq/5.2.5/Sidekiq#default_retries_exhausted=-class_method this method. It is available since [4.1.1](https://github.com/mperham/sidekiq/blob/master/Changes.md#411) so since nearly 3 years.

Reply:
We could use https://www.rubydoc.info/gems/sidekiq/5.2.5/Sidekiq#death_handlers-class_method on api side, but I don't see a way to use that from `PrometheusExporter::Instrumentation::Sidekiq`. 
There is also a worker-scoped retried_exhausted hook, but nothing that I see that we could use from inside the middleware :/

Reply:
Why can't you just call the method from the gem? Have you tried that?

Reply:
We could call https://www.rubydoc.info/gems/sidekiq/5.2.5/Sidekiq#default_retries_exhausted=-class_method from the middleware, but that method is deprecated :/